### PR TITLE
Polygon stuff

### PR DESCRIPTION
**Pull request by teseoch**

### CODE REVIEW

**Comment by teseoch** [comment]
the biggest change is here https://github.com/polyfem/polyfem/blob/37352911e3f3aac2c2a3bb678d8b599f76479532/src/polyfem/mesh/LocalBoundary.hpp

File: src/polyfem/state/StateInit.cpp [hunk 1]
```diff
@@ -296,7 +296,8 @@ namespace polyfem
 		}
 
 		// Save output directory and resolve output paths dynamically
-		const std::string output_dir = this->args["output"]["directory"];
+		const std::string output_dir = resolve_input_path(this->args["output"]["directory"]);
```
Comment (zfergus) [inline src/polyfem/state/StateInit.cpp:299]: Does this mean even the command line outout directory option has to be relative to the input JSON?

**Review by zfergus — APPROVED** [review]

File: src/polyfem/state/StateInit.cpp (hunk 1 above)
Comment (teseoch) [reply src/polyfem/state/StateInit.cpp:299]: yes, is this a bad thing?

Comment (zfergus) [reply src/polyfem/state/StateInit.cpp:299]: I like to specify -o outputdir/blah knowing the output will be placed in the current working directory. An easy fix is to resolve the relative path in the main so it becomes an absolute path.

Comment (teseoch) [reply src/polyfem/state/StateInit.cpp:299]: this will have a different effect if you use `-o` vs the stuff in the json, I like it

Comment (teseoch) [reply src/polyfem/state/StateInit.cpp:299]: fixed in 27e7a6dd5df537c3f54cc3bc20db9bbaad72f815

Comment (zfergus) [reply src/polyfem/state/StateInit.cpp:299]: Awesome, ship it!

**Pull request merged**